pkg/querier/vcs/token.go

```diff
@@ -140,18 +145,34 @@ func decodeToken(value string, key []byte) (*oauth2.Token, error) {
 
 	sessionToken := gitSessionTokenCookie{}
 	err = json.Unmarshal(decoded, &sessionToken)
-	if err != nil {
-		// This may be a legacy cookie. Legacy cookies aren't base64 encoded
-		// JSON objects, but rather a base64 encoded crypto hash.
-		var innerErr error
-		token, innerErr = decryptToken(value, key)
+	if err != nil || sessionToken.Token == nil {
+		// This may be a legacy cookie. Legacy cookies are base64 encoded legacyGitSessionTokenCookie objects.
+		token, innerErr := decodeLegacyToken(decoded, key)
```

Comment:
Not a blocker, but I think we can finally move away from this. It's safe to say that no one is using the OG session cookie and if they are, they're gonna have to re-auth. We also want to reduce the number "legacy" versions of this cookie we have floating around haha.

Reply:
In fact, I get rid of the legacy OG session data. 

Legacy here refers to the recently deprecated cookie that the PR address. This decode function now works with new format and has this backup plan for deprecated cookies. I will rename the function and rewrite the comment to be more clear!